go/vt/sqlparser/sql.y

```diff
@@ -433,13 +439,13 @@ select_expression:
   {
     $$ = &NonStarExpr{Expr: $1, As: $2}
   }
-| table_id '.' '*'
+| sql_id '.' '*'
```

Comment:
This and the other instance of AsTableIdent below (in column_name) are to avoid conflicts with things like function calls and other sql_ids